processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -106,7 +109,7 @@
     theBuffer.put(valueBytes.toByteArray());
     theBuffer.flip();
 
-    return new GenericIndexed<T>(theBuffer.asReadOnlyBuffer(), strategy, allowReverseLookup);
+    return new GenericIndexed<T>(theBuffer.asReadOnlyBuffer(), strategy, allowReverseLookup, version2);
```

Comment:
I have not tried it as this way lead to minimum changes to the code. I can try that if it has some benefits ?